src/shogun/kernel/LinearARDKernel.cpp

```diff
@@ -108,29 +106,24 @@ float64_t CLinearARDKernel::compute_helper(SGVector<float64_t> avec, SGVector<fl
 	}
 	else
 	{
-		left=SGMatrix<float64_t>(1,m_weights.num_rows);
-
 		SGMatrix<float64_t> ltmp(avec.vector,avec.vlen,1,false);
-
-		SGMatrix<float64_t> left_transpose(left.matrix,left.num_cols,1,false);
 		if(m_ARD_type==KT_DIAG)
-			linalg::elementwise_product(m_weights, ltmp, left_transpose);
+			left_transpose=linalg::elementwise_product(m_weights, ltmp);
 		else if(m_ARD_type==KT_FULL)
-			linalg::matrix_product(m_weights, ltmp, left_transpose);
+			left_transpose=linalg::matrix_product(m_weights, ltmp);
 		else
 			SG_ERROR("Unsupported ARD type\n");
+		left=SGMatrix<float64_t>(left_transpose.matrix,1,left_transpose.num_rows,false);
```

Comment:
@yorkerlin you need non-refcounted SGMatrix here because you're constructing it from a SGVector? I remember talking a couple of constructors in SGMatrix/SGVector to use it back and forth _with_ refcounting on. Maybe adding this now would be a good idea.